components/ActionSheet/tests/snapshot/MDCActionSheetItemTableViewCellSnapshotTests.m

```diff
@@ -0,0 +1,107 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialSnapshot.h"
+
+#import "../../src/private/MDCActionSheetItemTableViewCell.h"
+#import "MaterialActionSheet.h"
+
+/** Snapshot tests for @c MDCActionSheetItemTableViewCell. */
+@interface MDCActionSheetItemTableViewCellSnapshotTests : MDCSnapshotTestCase
+
+/** The cell that we take a snapshot of. */
+@property(nonatomic, strong, nullable) MDCActionSheetItemTableViewCell *cell;
+
+/** A @c MDCActionSheetAction used to create the cell we will test. */
+@property(nonatomic, strong, nullable) MDCActionSheetAction *action;
+
+@end
+
+@implementation MDCActionSheetItemTableViewCellSnapshotTests
+
+- (void)setUp {
+  [super setUp];
+
+  // Uncomment below to recreate all the goldens (or add the following line to the specific
+  // test you wish to recreate the golden for).
+  //  self.recordMode = YES;
+
+  self.cell = [[MDCActionSheetItemTableViewCell alloc] init];
+  self.action =
+      [MDCActionSheetAction actionWithTitle:@"Action 1"
+                                      image:[UIImage mdc_testImageOfSize:CGSizeMake(24, 24)]
+                                    handler:nil];
+}
+
+- (void)tearDown {
+  self.action = nil;
+  self.cell = nil;
+
+  [super tearDown];
+}
+
+- (void)generateSnapshotAndVerifyForView:(UIView *)view {
+  [self layoutCell];
+  UIView *snapshotView = [view mdc_addToBackgroundView];
+  [self snapshotVerifyView:snapshotView];
+}
+
+- (void)layoutCell {
+  self.cell.backgroundColor = UIColor.whiteColor;
+  self.cell.frame = CGRectMake(0, 0, 500, 75);
+  [self.cell setNeedsLayout];
+  [self.cell layoutIfNeeded];
+}
+
+- (void)testCellWithTitleAndAction {
+  // When
+  self.cell.action = self.action;
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.cell];
+}
+
+- (void)testCellWithTitleAndActionAndDivider {
+  // When
+  self.cell.action = self.action;
+  self.cell.showsDivider = YES;
+  self.cell.dividerColor = UIColor.greenColor;
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.cell];
+}
+
+- (void)testCellWithTitleOnlyAndDividerAndPositiveContentEdgeInsets {
+  // When
+  self.cell.action = self.action;
+  self.cell.showsDivider = YES;
+  self.cell.dividerColor = UIColor.greenColor;
+  self.cell.contentEdgeInsets = UIEdgeInsetsMake(0, 16, 0, 16);
+
+  // Then
+  [self generateSnapshotAndVerifyForView:self.cell];
+}
+
+- (void)testCellWithTitleOnlyAndDividerAndNegativeContentEdgeInsets {
+  // When
+  self.cell.action = self.action;
+  self.cell.showsDivider = YES;
+  self.cell.dividerColor = UIColor.greenColor;
+  self.cell.contentEdgeInsets = UIEdgeInsetsMake(0, -16, 0, -16);
```

Comment:
Tip: Set unbalanced values for `contentEdgeInsets` and include top/bottom to verify the behavior.  For example, `UIEdgeInsetsMake(-4, -8, -12, -16)` for negative values.  Those types of values would have exposed a layout bug in Chips that has been recently fixed.

Reply:
Done